### ISSUE TITLE
Consolidate Ivarator Configs

### ISSUE THREAD

**Issue by apmoriarty** [NationalSecurityAgency/datawave#2644]
It would be good to consolidate several ivarator configs together into a single config object. This would improve code readability and reduce the chance that a simple change in one area isn't propagated. 

As a first step, consolidate the following section of the ShardQueryLogic into a single IvaratorConfig object. For simplicity we can reuse the object mapper code from the IvaratorCacheDirConfig. 

For the sake of backwards compatibility and downstream integration, leave the getter and setter methods for each of these variables, but delegate them to the ivarator config object. 

```
    private List<IvaratorCacheDirConfig> ivaratorCacheDirConfigs = Collections.emptyList();
    private String ivaratorFstHdfsBaseURIs = null;
    private int ivaratorCacheBufferSize = 10000;
    private long ivaratorCacheScanPersistThreshold = 100000L;
    private long ivaratorCacheScanTimeout = 1000L * 60 * 60;
    private int maxFieldIndexRangeSplit = 11;
    private int ivaratorMaxOpenFiles = 100;
    private int ivaratorNumRetries = 2;
    private boolean ivaratorPersistVerify = true;
    private int ivaratorPersistVerifyCount = 100;
    private int maxIvaratorSources = 33;
    private long maxIvaratorSourceWait = 1000L * 60 * 30;
    private long maxIvaratorResults = -1;
```

**Comment by apmoriarty** [comment]
Future work could tackle the following

- Update QueryOptions to use an IvaratorConfig object instead of a bunch of individual query options
- Update IteratorBuildingVisitor to use an IvaratorConfig object instead of individual variables.
- Update each of the IvaratorBuilders to use an IvaratorConfig instead of individual variables.

**Comment by SethSmucker** [comment]
Should `private int maxIvaratorTerms = -1;` be included in the IvaratorConfig class as well?

**Comment by apmoriarty** [comment]
> Should `private int maxIvaratorTerms = -1;` be included in the IvaratorConfig class as well?

No, that variable is only used on the webserver side to enforce a limit on the number of ivarator terms in a query.

**Comment by SethSmucker** [comment]
Is the code snippet from `ShardQueryLogic` or `ShardQueryConfiguration`? I couldn't find it in the former but it was in the latter. I'm assuming it was SQC but I just wanted to double check.
